src/modules/status-list/status-list.service.ts

```diff
@@ -0,0 +1,204 @@
+import {
+  BadRequestException,
+  ConflictException,
+  Injectable,
+} from '@nestjs/common';
+import { URL } from 'url';
+import { ConfigService } from '@nestjs/config';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { verifyCredential } from 'didkit-wasm-node';
+import { gzip } from 'pako';
+import {
+  CredentialWithStatusDto,
+  StatusList2021EntryDto,
+} from './dtos/credential-status.dto';
+import { STATUS_LIST_MODULE_PATH } from './status-list.const';
+import { CredentialDto } from './dtos/credential.dto';
+import { VerifiableCredentialDto } from './dtos/verifiable-credential.dto';
+import { StatusListCredentialDto } from './dtos/status-list-credential.dto';
+import { StatusListVerifiableCredentialDto } from './dtos/status-list-verifiable-credential.dto';
+import { StatusListCredential } from './entities/status-list-credential.entity';
+import { DID } from '../did/did.types';
+
+@Injectable()
+export class StatusListService {
+  constructor(
+    private readonly configService: ConfigService,
+    @InjectRepository(StatusListCredential)
+    private readonly statusListRepository: Repository<StatusListCredential>
+  ) {}
+
+  /**
+   * Add StatusList2021Entry as `credentialStatus` parameter to given credential object.
+   * https://w3c-ccg.github.io/vc-status-list-2021/#statuslist2021entry
+   *
+   * @param {CredentialDto} credential
+   * @return credential with StatusList2021Entry
+   */
+  async addStatusListEntry(
+    credential: CredentialDto
+  ): Promise<CredentialWithStatusDto> {
+    if (await this.isCredentialRevoked(credential.id)) {
+      throw new ConflictException('Credential already revoked');
+    }
+
+    const statusListCredential = new URL(
+      `${STATUS_LIST_MODULE_PATH}/${credential.id}`,
+      this.configService.get('STATUS_LIST_DOMAIN')
+    );
+
+    const newStatusList = StatusListCredential.create({
+      credentialId: credential.id,
+      namespace: credential.credentialSubject.role.namespace,
+    });
+
+    await this.statusListRepository.save(newStatusList);
+
+    // https://w3c-ccg.github.io/vc-status-list-2021/#statuslist2021entrya
+    const credentialStatus: StatusList2021EntryDto = {
+      type: 'StatusList2021Entry',
+      statusPurpose: 'revocation',
+      statusListIndex: '0', // because we are using one-to-one mapping
+      statusListCredential: statusListCredential.href,
+    };
+
+    return { ...credential, credentialStatus };
+  }
+
+  /**
+   * Create a new StatusList2021Credential object for rejection purposes of a given credential for revoker.
+   * Object is not signed and not saved to database.
+   * https://w3c-ccg.github.io/vc-status-list-2021/#statuslist2021credential
+   *
+   * @param {VerifiableCredentialDto} credential
+   * @param {String} currentUser
+   * @return StatusList2021Credential without proof
+   */
+  async createStatusListCredential(
```

Comment:
A more accurate name, IMO
```suggestion
  async markStatusListCredential(
```

Reply:
Why `mark` is better?